cpp/cmake_modules/FindThrift.cmake

```diff
@@ -93,17 +93,22 @@ find_package_handle_standard_args(Thrift
                                   THRIFT_COMPILER)
 
 if(Thrift_FOUND OR THRIFT_FOUND)
-  set(Thrift_FOUND TRUE)
-  add_library(Thrift::thrift STATIC IMPORTED)
-  set_target_properties(Thrift::thrift
-                        PROPERTIES IMPORTED_LOCATION "${THRIFT_STATIC_LIB}"
-                                   INTERFACE_INCLUDE_DIRECTORIES "${THRIFT_INCLUDE_DIR}")
-  if(WIN32 AND NOT MSVC)
-    # We don't need this for Visual C++ because Thrift uses
-    # "#pragma comment(lib, "Ws2_32.lib")" in
-    # thrift/windows/config.h for Visual C++.
-    set_target_properties(Thrift::thrift PROPERTIES INTERFACE_LINK_LIBRARIES "ws2_32")
-  endif()
-
   extract_thrift_version()
+  if(FOUND_THRIFT_VERSION VERSION_LESS ${CMAKE_FIND_PACKAGE_NAME}_FIND_VERSION)
```

Comment:
We should not use our version check. We should use `find_package_handle_standard_args`'s `VERSION_VAR` feature: https://cmake.org/cmake/help/latest/module/FindPackageHandleStandardArgs.html

Example: https://github.com/apache/arrow/blob/master/cpp/cmake_modules/FindRapidJSONAlt.cmake#L59-L60